### ISSUE TITLE
CA1508: false positive inside catch block nested in for loop

### ISSUE THREAD

**Issue by innix** [dotnet/roslyn-analyzers#5160]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.301](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

I enabled all the analyzers on 1 of my projects and noticed this false positive:

```csharp
public static void Main()
{
    const int attempts = 5;

    for (int i = 0; i < attempts; i++)
    {
        try
        {
            Console.WriteLine($"Hello world: {i}");

            if (i == attempts - 1) // Last iteration.
            {
                throw new InvalidOperationException("Oops, something went wrong!");
            }
        }
        catch (Exception ex)
        {
            Console.WriteLine($"Exception caught: {ex.Message}");

            // CA1508: 'i == attempts - 1' is always 'false'. Remove or refactor the condition(s) to avoid dead code.
            if (i == attempts - 1)
            {
                Console.WriteLine("This should never happen, according to CA1508.");
            }
        }
    }
}
```

### Steps To Reproduce

Compile the above program with the following properties set in your `.csproj` file:

```xml
<PropertyGroup>
  <EnableNETAnalyzers>true</EnableNETAnalyzers>
  <AnalysisMode>AllEnabledByDefault</AnalysisMode>
</PropertyGroup>
```

### Expected behavior

`if (i == attempts - 1)` should not be flagged as dead code.

### Actual behavior

It gets flagged.

## Additional context

N/A.

**Comment by mavasani** [comment]
This still repros on the latest main branch sources.
